lib-driver/caqti_driver_sqlite3.ml

```diff
@@ -299,16 +299,33 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
         stmt: Sqlite3.stmt;
         row_type: 'b Caqti_type.t;
         query: string;
+        mutable affected_count: int;
+        mutable has_been_executed: bool;
       }
 
       let returned_count _ = return (Error `Unsupported)
 
-      let affected_count {stmt; _} = return (Error `Unsupported)
-      (* Might be implemented with Sqlite3.changes, but it needs to be queried
-       * at the correct time. *)
-
-      let fetch_row {stmt; row_type; query} =
-        (match Sqlite3.step stmt with
+      let affected_count {affected_count; has_been_executed; query; _} =
+        if has_been_executed
+        then return (Ok affected_count)
+        else
+          let msg =
+            Caqti_error.Msg
+              "Statement not executed yet, affected_count unavailable." in
+          return (Error (Caqti_error.response_rejected ~uri ~query msg))
+
+      let run_step response =
+        let ret = Sqlite3.step response.stmt in
+        if response.has_been_executed
+        then ret
+        else
+          (response.has_been_executed <- true;
+           response.affected_count <- Sqlite3.changes db;
+           ret)
+
+      let fetch_row response =
+        let {stmt; row_type; query; _} = response in
```

Comment:
Please use `response.*` notation below instead, or keep the original pattern match and alias with `as`.

Reply:
ok